packages/wdio-cli/src/commands/run.ts

```diff
@@ -209,47 +205,47 @@ export async function handler(argv: RunCommandArguments) {
      * Let's have WebdriverIO set it automatically if the user doesn't.
      */
     const nodePath = process.argv[0]
     let NODE_OPTIONS = process.env.NODE_OPTIONS || ''
     const isTSFile = wdioConf.fullPath.endsWith('.ts') || wdioConf.fullPath.endsWith('.mts')
     const runsWithLoader = (
         Boolean(
-            process.argv.find((arg) => arg.startsWith('--loader')) &&
-            process.argv.find((arg) => arg.endsWith('ts-node/esm'))
+            process.argv.find((arg) => arg.startsWith('--import') || arg.startsWith('--loader')) &&
+            process.argv.find((arg) => arg.endsWith('tsx'))
         ) ||
-        NODE_OPTIONS?.includes('ts-node/esm')
+        NODE_OPTIONS?.includes('tsx')
     )
     if (isTSFile && !runsWithLoader && nodePath) {
-        NODE_OPTIONS += ' --loader ts-node/esm/transpile-only --no-warnings'
-        if (nodeVersion('major') >= 20 || (nodeVersion('major') === 18 && nodeVersion('minor') >= 19)) {
-            // Changes in Node 18.19 (and up) and Node 20 affect how TS Node works with source maps, hence the need for this workaround. See:
-            // - https://github.com/webdriverio/webdriverio/issues/10901
-            // - https://github.com/TypeStrong/ts-node/issues/2053
-            NODE_OPTIONS += ' -r ts-node/register'
-        }
-        const tsNodeProjectFromEnvVar = process.env.TS_NODE_PROJECT &&
-            path.resolve(process.cwd(), process.env.TS_NODE_PROJECT)
-        const tsNodeProjectFromParams = params.autoCompileOpts?.tsNodeOpts?.project &&
-            path.resolve(process.cwd(), params.autoCompileOpts?.tsNodeOpts?.project)
-        const tsNodeProjectRelativeToWdioConfig = path.join(path.dirname(wdioConf.fullPath), 'tsconfig.json')
-        if (tsNodeProjectFromParams) {
-            console.log('Deprecated: use the TS_NODE_PROJECT environment variable instead')
+        // The `--import` flag is only available in Node 20.6.0 / 18.19.0 and later.
+        // This switching can be removed once the minimum supported version of Node exceeds 20.6.0 / 18.19.0
+        // see https://nodejs.org/api/module.html#customization-hooks
+        // and https://tsx.is/node#es-modules-only
+        const moduleLoaderFlag = (nodeVersion('major') >= 20 && nodeVersion('minor') >= 6) ||
+          (nodeVersion('major') === 18 && nodeVersion('minor') >= 19) ? '--import' : '--loader'
+        NODE_OPTIONS += ` ${moduleLoaderFlag} tsx`
+        const tsConfigPathFromEnvVar = process.env.TSX_TSCONFIG_PATH &&
+            path.resolve(process.cwd(), process.env.TSX_TSCONFIG_PATH)
+        const tsConfigPathFromParams = params.tsConfigPath &&
+            path.resolve(process.cwd(), params.tsConfigPath)
+        const tsConfigPathRelativeToWdioConfig = path.join(path.dirname(wdioConf.fullPath), 'tsconfig.json')
+        if (tsConfigPathFromParams) {
+            console.log('Deprecated: use the TSX_TSCONFIG_PATH environment variable instead')
```

Comment:
Let's just name this `TSCONFIG_PATH` as it isn't relevant to the user what tool we use to compile here.

Reply:
I did update the Typescript docs: https://github.com/webdriverio/webdriverio/pull/12752/files#diff-2d8355507a107fd3b03e5e7c41ffe62d8fb84241ad4dd70d5f8d374d072976b2

As for the `TSX_TSCONFIG_PATH`, this env var [comes from `tsx`](https://tsx.is/usage#custom-tsconfig-json-path).  I didn't add an env var version of the `tsConfigPath` config file property yet, as I thought we'd just use the `tsx` env var for this.  I think it would be a nicer user experience if we support both `TS_CONFIG_PATH` and `TSX_TSCONFIG_PATH`.

Reply:
Makes sense, let's support both and have this also documented.